sdk/monitor/azure-monitor-query/src/main/java/com/azure/monitor/query/LogsQueryAsyncClient.java

```diff
@@ -91,45 +94,63 @@
-     * @param logsQueryBatch {@link LogsQueryBatch} containing a batch of queries.
+     * @param logsBatchQuery {@link LogsBatchQuery} containing a batch of queries.
      * @return A collection of query results corresponding to the input batch of queries.@return
      */
     @ServiceMethod(returns = ReturnType.SINGLE)
-    public Mono<Response<LogsQueryBatchResultCollection>> queryLogsBatchWithResponse(LogsQueryBatch logsQueryBatch) {
-        return queryLogsBatchWithResponse(logsQueryBatch, Context.NONE);
+    public Mono<Response<LogsBatchQueryResultCollection>> queryLogsBatchWithResponse(LogsBatchQuery logsBatchQuery) {
+        return queryLogsBatchWithResponse(logsBatchQuery, Context.NONE);
     }
 
-    Mono<Response<LogsQueryBatchResultCollection>> queryLogsBatchWithResponse(LogsQueryBatch logsQueryBatch, Context context) {
-        BatchRequest batchRequest = new BatchRequest();
+    Mono<Response<LogsBatchQueryResultCollection>> queryLogsBatchWithResponse(LogsBatchQuery logsBatchQuery, Context context) {
         AtomicInteger id = new AtomicInteger();
 
-        List<LogQueryRequest> requests = logsQueryBatch.getQueries()
+        List<BatchQueryRequest> requests = logsBatchQuery.getQueries()
                 .stream()
                 .map(query -> {
                     QueryBody body = new QueryBody(query.getQuery())
-                            .setWorkspaces(query.getWorkspaceNames())
-                            .setAzureResourceIds(query.getAzureResourceIds())
-                            .setQualifiedNames(query.getQualifiedWorkspaceNames())
-                            .setWorkspaceIds(query.getWorkspaceIds());
-
-                    return new LogQueryRequest()
-                            .setId(String.valueOf(id.incrementAndGet()))
-                            .setBody(body)
-                            .setWorkspace(query.getWorkspaceId())
+                            .setWorkspaces(getAllWorkspaces(query));
+
+                    StringBuilder sb = new StringBuilder();
+                    if (query.isIncludeRendering()) {
+                        sb.append("include-render=true");
+                    }
+
+                    if (query.isIncludeStatistics()) {
+                        if (sb.length() > 0) {
+                            sb.append(",");
+                        }
+                        sb.append("include-statistics=true");
+                    }
+
+                    if (query.getServerTimeout() != null) {
+                        if (sb.length() > 0) {
+                            sb.append(",");
+                        }
+                        sb.append("wait=");
+                        sb.append(query.getServerTimeout().getSeconds());
+                    }
+
+                    String preferHeader = sb.toString().isEmpty() ? null : sb.toString();
+
+                    Map<String, String> headers = new HashMap<>();
+                    headers.put("Prefer", preferHeader);
```

Comment:
>preferHeader [](http://example.com/codeflow?start=42&length=12)

What happens if you end the preferHeader with a ','? If it is a no-op, you can directly do something like sb.append(include-statistics=true,) and not have to do if check for length and the second append.

Reply:
I think it shouldn't matter but it's better to keep it well-formatted as defined in the RFC for prefer headers.